dashboard/app/controllers/lessons_controller.rb

```diff
@@ -39,7 +39,11 @@ def update
       @lesson.update_objectives(JSON.parse(params[:objectives])) if params[:objectives]
     end
 
-    redirect_to lesson_path(id: @lesson.id)
+    if params[:do_not_redirect]
+      render json: @lesson
+    else
+      redirect_to lesson_path(id: @lesson.id)
```

Comment:
why do we still need the redirect case to work?

Reply:
for the normal saving case right?

Reply:
ahh, I see. I think it is important to do the post/put via ajax in the "Save and Close" case as well, and only navigate away after a successful save. This helps us preserve any pending edits if the fail saves for a specific reason which can be corrected by the user, which looks like it is captured as [PLAT-422](https://codedotorg.atlassian.net/browse/PLAT-422). Another benefit of doing this is that we will not have to keep maintaining two ways of saving things, which could lead to weird bugs, for example an input field that is named incorrectly might work for saving via ajax but not via form post. Does this make sense? I'm having trouble remembering if there is another consideration here so please feel free to call out anything I might be missing.

Reply:
Interesting. Does that mean we would do the navigation on success of the ajax call client side instead of server side?

Reply:
correct.

Reply:
sometimes the `navigateToHref` helper is used to make the navigation unit-testable.

Reply:
Alright I think have created a follow up PR that accomplishes this: https://github.com/code-dot-org/code-dot-org/pull/37802

Reply:
a separate PR sounds good!